src/extraction/FStar.Extraction.ML.Term.fs

```diff
@@ -170,11 +170,17 @@ let rec is_arity env t =
     | Tm_type _ -> true
     | Tm_arrow(_, c) ->
       is_arity env (FStar.Syntax.Util.comp_result c)
-    | Tm_fvar _ ->
-      let t = N.normalize [Env.AllowUnboundUniverses; Env.EraseUniverses; Env.UnfoldUntil delta_constant] env.env_tcenv t in
-      begin match (SS.compress t).n with
-        | Tm_fvar _ -> false
-        | _ -> is_arity env t
+    | Tm_fvar fv ->
+      let topt =
+        FStar.TypeChecker.Env.lookup_definition
+          [Env.Unfold delta_constant]
```

Comment:
Would it unfold definitions that are marked `inline_for_extraction`? I am looking at `Env.visible_at`. There, with `d` being `Unfold delta_constant` -- which is what is being passed here -- and `q` being `Inline_for_extraction`, `visible_at` returns `false`. Am I understanding it incorrectly?